pkg/controller/core.oam.dev/v1alpha2/application/application_controller.go

```diff
@@ -216,6 +216,19 @@ func (r *Reconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctrl.Resu
 		return r.gcResourceTrackers(logCtx, handler, common.ApplicationRendering, false)
 	case common.WorkflowStateSuspended:
 		logCtx.Info("Workflow return state=Suspend")
+		doWaiting, durationWaiting, err := wf.HandleSuspendWait(logCtx)
+		if err != nil {
+			return r.endWithNegativeCondition(logCtx, app, condition.ErrorCondition(common.WorkflowCondition.String(), err), common.ApplicationRunningWorkflow)
+		}
+		if doWaiting {
+			if durationWaiting > 0 {
+				_, err = r.gcResourceTrackers(logCtx, handler, common.ApplicationWorkflowSuspending, false)
+				return r.result(err).requeue(durationWaiting).ret()
+			}
+			handler.app.Status.Workflow.Suspend = false
+			handler.app.Status.Workflow.SuspendState = ""
+			return r.result(r.updateStatus(ctx, handler.app, common.ApplicationRunningWorkflow)).ret()
```

Comment:
Please use `r.gcResourceTrackers` here, you can add `isPatch` in this function to check whether to use update or patch in `r.gcResourceTrackers`.